### PR TITLE
EOS-23002: Node Health - Set resource health status 'FAULT' if it is 'NA'  - Resource health status is coming as 'NA' for 'inactive' services and other resources (CPU, SAS_PORTS)

### PR DESCRIPTION
**Pull request by mariyappanp**
## Problem Statement:

Node Health - Resource health status is coming as 'NA' for 'inactive' services and other resources (CPU, SAS_PORTS)

## Solution Design:

Set resource health status 'FAULT' if it is 'NA'

## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [x] Unittest updated and all unittests are passing?

* [x] Sanity/Self tests are updated (if applicable)?

* [x] Manual testing done covering happy path and non-happy path?

## Integration:

* [x] If there is any interface change, did you communicate it to other components?
* [x] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [x] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def get_cpu_info(self, add_overall_usage=False):
             uid = f"CPU-{cpu_id}"
             cpu_dict = self.get_health_template(uid, is_fru=False)
             online_status = "Online" if cpu_id in cpu_online else "Offline"
-            health_status = "OK" if online_status == "Online" else "NA"
-            usage = "NA" if health_status == "NA" \
+            health_status = "OK" if online_status == "Online" else "FAULT"
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py:253]: This is wrong, we cant default to Fault health state if health is not available. NA states we cant determine health state of given resource, like here if CPU is offline, same case with inactive service/s ..

Same applies for all other changes here

**Review by cdeshmukh — COMMENTED** [review]
We dont changes at SSPL end depending on what i understood till now

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py [hunk 2]
```diff
@@ -88,6 +88,9 @@ def set_health_data(health_data: dict, status, description=None,
         if not recommendation:
             recommendation = 'NA' if good_state\
                 else DEFAULT_RECOMMENDATION
+        else:
+            if "No action is required" in recommendation.lower():
```
[the remark this answers is not in the document]
Comment (sumedhak27) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/resource_map.py:92]: The first letter `N` is capital and we are comparing with `recommendation.lower()` ??
I think, either N should be small or we can use `"No action is required".lower()`.

File: low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py (hunk 1 above)
Comment (jjxsg) [reply low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py:253]: I think Node CLI should have another status as `NA` apart from `FAULT`, `DEGRADED` and `OK`

**Comment by mariyappanp** [comment]
As we don't need changes on SSPL side and Provisioner will fix this in Node CLI. Hence closing this.

**Pull request closed without merging**